### DISCUSSION TITLE
AWS cannot establish connection to Athena

### DISCUSSION THREAD

**Discussion by ppruchnicki** [dbeaver/dbeaver#14617] [Q&A] [answered]
#### System information: 
- Operating system (distribution) and version: Linux Ubuntu 20.04.3 LTS
- DBeaver version:  21.2.2.202110030714

#### Connection specification:
- Database name and version: Athena
- Driver name: AthenaJDBC42.jar
- Do you use tunnels or proxies (SSH, SOCKS, etc)? No

#### Describe the problem you're observing:
I'm using AWS Azure Login to authenticate to AWS services and I want to connect to Athena through DBeaver, unfortunately I come up against this error 
[Simba][AthenaJDBC](100121) An error has occurred.
  profile file cannot be null

#### Steps to reproduce, if exist:
As it is described in the instruction I set the correct region and S3 location - I've triple checked that - Username and password is filled with no important values since it's not used at all. I fill neccessary driver properties like AwsCredentialsProviderArguments with the profile name I created, AwsCredentialsProviderClass with `com.simba.athena.amazonaws.auth.profile.ProfileCredentialsProvider` and workgroup with my workgroup from AWS. After authenticating using AWS Azure Login, the config file is set up correctly, same goes with credential file, I can see my profile name with all atributes. It seems like DBeaver cannot locate the credential file. Of course it is present in ~/.aws .

#### Include any warning/errors/backtraces from the logs
[Simba][AthenaJDBC](100121) An error has occurred.
  profile file cannot be null

**Comment by Matvey16** [comment]
@ppruchnicki Hello, can you specify what do you mean by AWS Azure Login? Is it some kind of utilty for authentication in AWS?

**Comment by ppruchnicki** [comment]
@Matvey16  Yes, it is used to authenticate with AWS using Active Directory https://github.com/sportradar/aws-azure-login

**Comment by Matvey16** [comment]
@ppruchnicki  What type of DBeaver package do you use? Installing from Ubuntu software or with .deb package form our site or maybe with apt-get?

**Comment by ppruchnicki** [comment]
@Matvey16 I used snap installed from Ubuntu software: 
dbeaver-ce               21.2.2.202110030714         147    latest/stable    dbeaver-corp   -
Installed using this command line:
sudo snap install dbeaver-ce

**Comment by Matvey16** [comment, marked as answer]
@ppruchnicki I see. That is a problem with snap version. This package doesn't have permissions to read ~/.aws directory. That issue is being fixed right now, but is not ready yet. As a temporary workaround you could install DBeaver from https://dbeaver.io/download/

**Comment by ppruchnicki** [comment]
I can confirm switching to another instance of DBeaver worked for me. Thank you for the workaround.
